### PR TITLE
feat(app): Dark Theme, Brand Color Update & Advanced Colouring

### PR DESCRIPTION
**Pull request by fabric-8**
# Description

- Sync colors with Design
- Update brand color
- Make base colors more abstract
- Introduce Themability (Mostly via silent classes and mixins living in a dedicated sass file)
- Create Dark theme, refactor colorization

# Motivation and Context

Dark Mode and general desirability to add more themes in the future.

# Screenshots (if appropriate)

Kapture 2019-04-26 at 22 42 45

# Types of changes
- [ ] Chore (tests, refactors, and fixes)
- [x] New feature (adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# TODOS

### General

> *What do we do with the splash screen animation? Does the animation need to change to the new color?*

> *The app icon in the taskbar is still the original color*

**I propose to leave both for now. Other apps I'm aware of do it in a similar way and only apply the theming to everything "inside" the actual app but not the "peripherals"**

- [x] Update app icon & preloader colors
- [x]  The AppStore category titles are hardly readable (unless this is intended, but on my screen the contrast isn't great)
- [x]  The dropdown section still doesn't have a "dark" background - but this actually doesn't bother me that much, so this can remain optional if that seems better suited (changing it may bring up a lot of hassle)
- [x]  Remove focus outline around theme toggle
- [x]  In dark theme, the node status circle has a white outline
    - Had to rebuild the icon in a slightly different way - We're now theme-independent here 👍
- [ ]  In dark theme, the NOS token icon reveals something white

This should be fixed as soon as the new asset has been uploaded

- [x]  In dark theme, the sidebar has a light colored outline, it looks slightly "intrusive", perhaps it could be lower contrast?

---

### Wallet

- [x]  The icon in the holdings/send tokens list is still the original color
- [x]  The % text of the donut is not readable (you can see this with the account file I'll provide)
- [x]  The history items are not yet styled
- [x]  The Not Using GAS Priority Fee | Change on the send tokens panel --> the | stands out (lol little nitpick sorry haha!)
- [x] Adjust Skeleton Loader Animation

---

### Login

- [x] The secret pill is not adjusted
- [x] When hitting "Login" while signed in, text labels have a darker background
- [x]  Dark themed register screen (first one) has a weird input field around the checkbox to use the Ledger. I might have to change this perhaps? Shall I make a "new" component for it - or are you able to style it using the existing underlying button
- [x]  On the dark themed Login screen, the "secret word" pill is not styled
- [x]  The ✍️ Verify your recovery seed from the last step in the last step of registration seems on the background/pushed in. IMO it should pop out as it's a "header" (up to you to decide tho)
- [x]  The "Log In" text at the "Already Registered? Log In" footer could be bold
- [x]  Copy to Clipboard the copy notification is under the modal backdrop z-index
    - Fixed. Also adjusted colors to make it clearly stand out no matter light or dark

> The blue gradient side panel on the register screens doesn't really fit the dark theme IMO - cc @deanpress your opinion?

Tbh I think it's fine for now — Was planning to vamp this up more in the future, suggesting to look into this again when maybe creating some visuals for that space

---

### Account Management

- [x]  The account management titles are hardly readable
- [x]  The account management bars are not yet styled
    - Adjusted the styling towards better readability and introduced breakpoints as well

### CODE REVIEW

**Comment by DalderupMaurice** [comment]
Amazing work man!  

Couple questions/notices:
- Is there no dark version needed for the app store cards? 
- The app store category titles might be a little dark? 
- I Almost can't see a difference in colors between the normal/hovered state of the minimize/maximize screen buttons
- When switching between tabs (nash.io and the app store) the "disabled" state of the forward/backward and refresh buttons look brighter than the enabled state. Is this intentional? 
-  In the popup when you click "Add custom network" in settings, the title is not readable
- Should the dropdown have a colored BG as well? (try to select a currency in the settings tab)
- The hover-state of the settings menu items are not readable [](https://i.imgur.com/WcFTW44.png)

It's nice that you've added in the new BG for the dark theme :P 
I'll try to finish the login/register stuff today - in which case the TODO"s can be completed

**Comment by fabric-8** [comment]
@DalderupMaurice Thanks a lot for the feedback!

Covered all of the things you mentioned and continued with theming, bringing everything to a state which I think seems close to prime-time ready :)

There are still small systematic things that can be improved (use vars EVERYWHERE etc), but I think those things can be improved and fixed over time.

Maybe everybody can have another look for a final round of improvements and fixes (which I'd aim to do over the upcoming weekend)

**Comment by DalderupMaurice** [comment]
Awesome! Got some feedback left. I'll write it down here and shoot you a private message with an account file & some screenshots.

The account file will contain a profile with a transaction history, as these are not yet covered.

Some leftover items:
- What do we do with the splash screen animation? Does the animation need to change to the new color?
- The app icon in the taskbar is still the original color
- The icon in the holdings/send tokens list is still the original color
- The % text of the donut is not readable (you can see this with the account file I'll provide)
- The history items are not yet styled
- The account management bars are not yet styled
- The account management titles are hardly readable
- The AppStore category titles are hardly readable (unless this is intended, but on my screen the contrast isn't great)
- The `Not Using GAS Priority Fee | Change` on the send tokens panel --> the | stands out (lol little nitpick sorry haha!) 
- The dropdown section still doesn't have a "dark" background - but this actually doesn't bother me that much, so this can remain optional if that seems better suited (changing it may bring up a lot of hassle)
- Dark themed register screen (first one) has a weird input field around the checkbox to use the Ledger. I might have to change this perhaps? Shall I make a "new" component for it - or are you able to style it using the existing underlying button
- On the dark themed Login screen, the "secret word" pill is not styled
- The blue gradient side panel on the register screens doesn't really fit the dark theme IMO - cc @deanpress your opinion?
- The `✍️ Verify your recovery seed from the last step` in the last step of registration seems on the background/pushed in. IMO it should pop out as it's a "header" (up to you to decide tho) 

These are the final things I see - and after these are done we're most likely 100% through it! :) 🎉 💯 🔥 🥇

**Comment by deanpress** [comment]
Super smooth! Love using the client with these new colors and dark theme.

Here are some points of feedback:

* The app store grid messes up with smaller body width (by extending console panel. there's still a min width for the actual window):

image

* The icon on Windows seems pixellated (not with 0.5.4, only this PR):

image

* The mnemonic words are oddly positioned, and if you click Copy to Clipboard the copy notification is under the modal backdrop z-index:

image

* There's a button outline when you click Enable Dark Theme:
image

* In dark theme, the node status circle has a white outline:
image

* In dark theme, the NOS token icon reveals something white:
image

* The Wallet window always has a y-axis scroll.

* In Dark Theme the Key Management window's style is not updated (I also think the elements and font should be scaled relative to window size):
image

* In dark theme, the sidebar has a light colored outline, it looks slightly "intrusive", perhaps it could be lower contrast?
image

* There's zero visual feedback when clicking Logout.

* If you already have an account and you click Log In (or the Wallet), you should go to the Log In window instead of the New Wallet window.

* In dark theme, the "Use ledger for next step" checkbox is in a styled container. I think the container can just be transparent without outline:
image

I agree with @DalderupMaurice that the blue "Steps" container on the account window could have a different color to better complement the dark theme. It looks fine in the light theme though imo. The "Log In" text at the "Already Registered? Log In" footer could be bold.

image

**Comment by DalderupMaurice** [comment]
cc @deanpress @fabric-8 

Some comment/clarification on @deanpress his comments 

`The app store grid messes up with smaller body width (by extending console panel. there's still a min width for the actual window)`
--> This is only when console is open. It scales well when it is closed

`The mnemonic words are oddly positioned, and if you click Copy to Clipboard the copy notification is under the modal backdrop z-index`
--> I didn't knew how to properly position the words and used flex - space-between to even out the edges. It's hard to make it show properly as each time the words in length (try going back and forth a couple times and you see what I mean). I don't know if @fabric-8 knows a way to make it work every time? 
Regarding the toast - It's been a problem I can't find for quite some while. Not sure why it's doing that. z-index doesn't fix it

`In dark theme, the NOS token icon reveals something white:`
--> Needs to be replaced with new color

`There's zero visual feedback when clicking Logout.`
--> Would need a design - if it includes any logic I can take this on me

`If you already have an account and you click Log In (or the Wallet), you should go to the Log In window instead of the New Wallet window.`
--> This logic is already included - but commented out for easy development purposes. I'll re-enable this in my PR as we're going to merge this branch in there aswell

**Comment by fabric-8** [comment]
Thanks a bunch for the feedback everybody! This provides a solid list to go through to wrap things up ❤️

**Comment by fabric-8** [comment]
@deanpress @DalderupMaurice Just pushed a larger update – I've added all "tasks" relevant to this PR to the [first post / description](https://github.com/nos/client/pull/970#issue-274051625). 

@deanpress You mentioned a few things which I think would be better tackled separately as this PR here become quite a beast already :D 

- There's zero visual feedback when clicking Logout
- If you already have an account and you click Log In (or the Wallet), you should go to the Log In window instead of the New Wallet window

We can tackle them as seperate design tasks (& PRs)

**Comment by DalderupMaurice** [comment]
> @deanpress @DalderupMaurice Just pushed a larger update – I've added all "tasks" relevant to this PR to the [first post / description](https://github.com/nos/client/pull/970#issue-274051625).
> 
> @deanpress You mentioned a few things which I think would be better tackled separately as this PR here become quite a beast already :D
> 
> * There's zero visual feedback when clicking Logout
> * If you already have an account and you click Log In (or the Wallet), you should go to the Log In window instead of the New Wallet window
> 
> We can tackle them as seperate design tasks (& PRs)

Agreed. My PR is also WAYYYY too big 😂 I'm leaving some things out as well

File: src/common/stylesheets/variables.scss [hunk 1]
```diff
@@ -26,3 +26,6 @@ $primary-border-style: thin solid rgba(0, 0, 0, 0.05);
 
 // Responsive
 $responsive-width: 1024px;
+
+// Sizes
+$containerSize: 1024px;
```
Comment (DalderupMaurice) [inline src/common/stylesheets/variables.scss]: Does this mean we can dynamically adjust all the "root" containers? 
We've had some discussions as to where the 1024px was WAY too small on high res screens :P

**Comment by deanpress** [comment]
>browse to docs.nos.io and see the magic!!! It actually turns dark/white together with the nOS Client 😂
@DalderupMaurice I think the body has no background color, so you see the background of the parent container.

File: src/common/stylesheets/variables.scss (hunk 1 above)
Comment (fabric-8) [reply src/common/stylesheets/variables.scss]: Yes sure, really just a max-width value which we can easily change and adjust

**Comment by fabric-8** [comment]
@DalderupMaurice @deanpress Ok I think I have everything!

There's an issue when logging in: Even when I have multiple accounts, I can't open the dropdown :/ (Doesn't do anything when clicking). Any idea what happened there? Any external factors influencing this?

**Comment by DalderupMaurice** [comment]
Awesome @fabric-8! IMO everything looks good :) 
I'll resolve the merge conflicts later today and get it going!

**Comment by DalderupMaurice** [comment]
> @DalderupMaurice @deanpress Ok I think I have everything!
> 
> There's an issue when logging in: Even when I have multiple accounts, I can't open the dropdown :/ (Doesn't do anything when clicking). Any idea what happened there? Any external factors influencing this?

I'll try to figure it out, it's non-blocking for this PR tho, as it's a functional issue

**Comment by DalderupMaurice** [comment]
Thanks! @fabric-8 ❤️

**Comment by DalderupMaurice** [comment]
I'll keep this branch and the `feat/dark-theme` branch until everything is finished :)

**Pull request merged**